### PR TITLE
[CIS-1297] CocoaPods + SwiftPM OSS integration + Carthage

### PR DESCRIPTION
**Pull request by bielikb**
### 🔗 Issue Link

[CIS-1297](https://stream-io.atlassian.net/browse/CIS-1297)

### 🎯 Goal

Provide integration apps

### 🛠 Implementation

* Verify CocoaPods integration (OSS)
* Verify SwiftPM integration (OSS)
* Added project + scripts for Carthage

### 🧪 Testing

N/A

### 🎨 Changes

* New `StreamChatIntegration` workspace that contains projects for CocoaPods + SwiftPM with shared source code
* Adds scripts/workflow to make CI happy. Jobs that verify the integrations are run in matrix
* Adds templates/other git/github related setup that we leverage in our own repo

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Uncomment the next line to define a global platform for your project
+platform :ios, '11.0'
+
+workspace 'StreamChatIntegration'
+
+target 'StreamChatIntegration-CocoaPods' do
+  project 'StreamChatIntegration-CocoaPods.xcodeproj'
+  pod 'StreamChatUI'
```
Comment (polqf) [inline Podfile:8]: Are we going to be using this for PR merges into main as well? If that's the case, we would need to pass the branch

File: StreamChatIntegrationSource/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
+//
+//  AppDelegate.swift
+//  StreamChatIntegration-SPM
+//
+//  Created by Boris Bielik on 16/11/2021.
+//
+
+import UIKit
```
Comment (polqf) [inline StreamChatIntegrationSource/AppDelegate.swift:8]: I'm guessing that this file is used by all the integration projects, right?

File: StreamChatIntegrationSource/Base.lproj/Main.storyboard [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (polqf) [inline StreamChatIntegrationSource/Base.lproj/Main.storyboard:1]: Can we remove the storyboards? Less xml, less conflicts 😄

File: StreamChatIntegrationSource/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+//  ViewController.swift
+//  StreamChatIntegration-SPM
+//
+//  Created by Boris Bielik on 16/11/2021.
+//
+
+import UIKit
+import StreamChatUI
+
+class ViewController: ChatChannelVC {
```
Comment (polqf) [inline StreamChatIntegrationSource/ViewController.swift:11]: You can remove this file, right?

Comment (bielikb) [reply StreamChatIntegrationSource/ViewController.swift]: Thanks for spotting this. File is removed :)

File: StreamChatIntegrationSource/Base.lproj/Main.storyboard (hunk 3 above)
Comment (bielikb) [reply StreamChatIntegrationSource/Base.lproj/Main.storyboard]: Removed :)

File: StreamChatIntegrationSource/AppDelegate.swift (hunk 2 above)
Comment (bielikb) [reply StreamChatIntegrationSource/AppDelegate.swift:5]: That's correct. All integrations do shared 1 source code.

File: Podfile (hunk 1 above)
Comment (bielikb) [reply Podfile:8]: For now we can keep it this way. If we will need to point the pods towards some specific branch during our development time we can do so, once the setup for binary frameworks will be in place

**Pull request closed without merging**